src/capwap_http_api_v2_handler.erl

```diff
@@ -0,0 +1,103 @@
+%% the Free Software Foundation, either version 3 of the License, or
+%% (at your option) any later version.
+
+%% This program is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%% GNU Affero General Public License for more details.
+
+%% You should have received a copy of the GNU Affero General Public License
+%% along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+-module(capwap_http_api_v2_handler).
+
+-export([handle/4]).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+handle(<<"GET">>, [<<"station">>], Req, State) ->
+	Stations0 = capwap_station_reg:list_stations(),
+	Stations1 = lists:foldl(fun({AC, MAC}, Acc) ->
+					orddict:update(AC, fun(V) -> [MAC|V] end, [MAC], Acc)
+				end, orddict:new(), Stations0),
+	Stations = lists:map(fun({K, V}) ->
+					[WTP|_] = capwap_wtp_reg:get_commonname(K),
+				{WTP, V}
+			     end, orddict:to_list(Stations1)),
+	Res = lists:foldl(fun format_wtps/2, [], Stations),
+	{jsx:encode(Res), Req, State};
+
+handle(<<"DELETE">>, [<<"wtp">>, _WTP, <<"station">>, Station], Req, State) ->
+	case capwap_tools:mac_to_hex(Station) of
+		MACSta when is_binary(MACSta) ->
+			case capwap:detach_station(MACSta) of
+				ok ->
+					{true, Req, State};
+				_ ->
+					{false, Req, State}
+			end;
+		_ ->
+			{false, Req, State}
+	end;
+
+handle(_Method, _Path, Req, State) ->
+	{jsx:encode([{error, bad_command}]), Req, State}.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+format_wtps({{WTP, _WtpIP}, STAs}, Acc0) ->
+	Radios = lists:foldl(fun format_radios/2, #{}, STAs),
+
+	ResRadios = maps:fold(fun(RadioId, STAs0, Acc) ->
+				[[{'radio-id', RadioId},
+				  {stations, STAs0}] | Acc]
+			      end, [], Radios), [[{'wtp-id', WTP},
+						  {radios, ResRadios}] | Acc0].
+
+format_radios(ClientMAC, Acc) ->
+	case capwap_station_reg:lookup(ClientMAC) of
+		{ok, Pid} ->
+			{ok, RSSI} = ieee80211_station:get_rssi(Pid),
+			case get_station_statistic(ClientMAC) of
+				{RadioId, Stats} ->
+					update_radio(RadioId,  Acc,
+						     [
+						      {mac, bin_fmt("~s", [capwap_tools:format_mac(ClientMAC)])},
+						      {rssi, RSSI}
+						     ] ++ Stats );
+				_ -> Acc
+			end;
+		_ ->
+			Acc
+	end.
+
+get_station_statistic(MAC) ->
+	case capwap_dp:get_station(MAC) of
+		{_MAC, VLan, RadioId, _BSS, {RcvdPkts, SendPkts, RcvdBytes, SendBytes}} ->
+			{RadioId, [{vlan, VLan},
+				   {'rx-packets', RcvdPkts},
+				   {'rx-bytes', RcvdBytes},
+				   {'tx-packets', SendPkts},
+				   {'tx-bytes', SendBytes}]};
+		_ -> undefined
+	end.
+
+update_radio(Id, Acc, Station) ->
```

Comment:
can be done much simpler with:
```maps:update_with(Id, fun(Stations) -> [Station|Stations] end, [Station], Acc).```

If you use the `cut` parse_transform this even becomes:
```maps:update_with(Id, [Station|_] end, [Station], Acc).```